### PR TITLE
fix: make hard-coded HTTP "insecure" options configurable

### PR DESCRIPTION
**Pull request by amal-thundiyil**
re #2216

### CODE REVIEW

File: cmd/reva/main.go [hunk 1]
```diff
@@ -114,10 +114,8 @@ func main() {
 	}
 
 	client = rhttp.GetHTTPClient(
-		// TODO make insecure configurable
-		rhttp.Insecure(true),
-		// TODO make timeout configurable
-		rhttp.Timeout(time.Duration(24*int64(time.Hour))),
+		rhttp.Insecure(insecure),
```
Comment (ishank011) [inline cmd/reva/main.go:117]: This `insecure` flag denotes if the GRPC connection to the gateway should work without TLS or not. I'm not sure if we should use the same one for the data gateway service.

**Review by ishank011 — COMMENTED** [review]
Hi @amal-thundiyil, I left a couple of comments

File: cmd/reva/download.go [hunk 2]
```diff
@@ -104,10 +104,8 @@ func downloadCommand() *command {
 			httpReq.Header.Set(datagateway.TokenTransportHeader, p.Token)
 			httpClient := rhttp.GetHTTPClient(
```
Comment (ishank011) [inline cmd/reva/download.go:105]: Let's use the `client` declared in the `main.go` file instead of a new one.

**Review by ishank011 — COMMENTED** [review]
Hi @amal-thundiyil. One minor comment, we can merge then

File: cmd/reva/download.go [hunk 3]
```diff
@@ -38,7 +37,7 @@ import (
 	"github.com/studio-b12/gowebdav"
 )
 
-func downloadCommand() *command {
+func downloadCommand(httpClient *http.Client) *command {
```
Comment (ishank011) [inline cmd/reva/download.go:40]: `client` is a global variable, you don't need to pass it as a function argument

Comment (amal-thundiyil) [reply cmd/reva/download.go]: Ah, sorry. Fixed it.

**Pull request merged**